### PR TITLE
Feature theme improvement

### LINKED ISSUE #119: Fix theme coherence

### PR DESCRIPTION
**Pull request by andrepeiry**
This pull request contains major theme improvements. Most of Material 3 color have been defined (outside *scrim*, *outline* and *outlineVariant*) for both a light and dark theme. Some colors might still be discussed, but the main work is here.
AddParking incomplete  example
Some of the element on this example are not correct, due to the fact that this PR is focus on the theme, and that the Atom and Molecule are not yet refresh and implemented everywhere. But this will be done in another PR.

A nice `ThemePreview` has been added, to quickly visualize the theme color, as well as most of the components.
Light Theme PreviewDark Theme Preview

The logo has been reworked as an SVG, and provided in multiple dimension for the `drawable/mipmap` directory.
Reworked Cyrcle Logo
The loading logo (and the one display on the "desktop" of the phone are displayed correctly, might need more investigation.

Code Coverage as of rebasing the branch onto commit `4b86a52b` "Merge pull request #116 from SwEnt-Fall-2024-Group-22/refactor-test-files" :
image
Code Coverage Report

Closes #119

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/cyrcle/ui/theme/molecules/BottomNavigationBar.kt
Comment (matttsss) [inline app/src/androidTest/java/com/github/se/cyrcle/ui/theme/molecules/BottomNavigationBar.kt:1]: You should maybe add `Test` to the filename

Comment (matttsss) [inline app/src/androidTest/java/com/github/se/cyrcle/ui/theme/molecules/BottomNavigationBar.kt:1]: You should maybe add `Test` to the filename

**Review by matttsss — APPROVED** [review]
Good changes, thanks for the new logo!

File: app/src/androidTest/java/com/github/se/cyrcle/ui/theme/ColorTest.kt
Comment (LouisJamesVasseur) [inline app/src/androidTest/java/com/github/se/cyrcle/ui/theme/ColorTest.kt:1]: Tests seem nice and complete, great work!

File: app/src/androidTest/java/com/github/se/cyrcle/ui/theme/ThemeTest.kt
Comment (LouisJamesVasseur) [inline app/src/androidTest/java/com/github/se/cyrcle/ui/theme/ThemeTest.kt:1]: Solid tests here too!

File: app/src/main/res/drawable/app_logo_name.png
Comment (LouisJamesVasseur) [inline app/src/main/res/drawable/app_logo_name.png:1]: Very nice logo!

**Review by LouisJamesVasseur — APPROVED** [review]
A great rework! Happy to see so many improvements/ added test coverages across the board!

**Pull request merged**